test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/deregistration/DeregisterReasonControllerSpec.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.controllers.deregistration
+
+import base.unit.ControllerSpec
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, verify, when}
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.data.Form
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.json.JsObject
+import play.api.test.Helpers.{redirectLocation, status}
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.deregistration.DeregisterReasonForm
+import uk.gov.hmrc.plasticpackagingtax.registration.models.deregistration.DeregistrationReason.CeasedTrading
+import uk.gov.hmrc.plasticpackagingtax.registration.models.deregistration.{
+  DeregistrationDetails,
+  DeregistrationReason
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.repositories.{
+  DeregistrationDetailRepositoryImpl,
+  DeregistrationDetailsRepository,
+  UserDataRepository
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.deregistration.deregister_reason_page
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+import scala.concurrent.Future
+
+class DeregisterReasonControllerSpec extends ControllerSpec {
+  private val page       = mock[deregister_reason_page]
+  private val mcc        = stubMessagesControllerComponents()
+  private val mockCache  = mock[UserDataRepository]
```

Comment:
There's a `MockDeregistrationDetailRepository` trait which defines `inMemoryDeregistrationDetailRepository` which might have been used instead of having to create your own mock.

Reply:
Yes, I did see that but I prefer to mock these kind of things rather than use a real implementation.